crates/provider-blobstore-nats/src/config.rs

```diff
@@ -0,0 +1,443 @@
+
+const DEFAULT_NATS_URI: &str = "nats://0.0.0.0:4222";
+
+const CONFIG_NATS_URI: &str = "cluster_uri";
+const CONFIG_NATS_JETSTREAM_DOMAIN: &str = "js_domain";
+const CONFIG_NATS_CLIENT_JWT: &str = "client_jwt";
+const CONFIG_NATS_CLIENT_SEED: &str = "client_seed";
+const CONFIG_NATS_TLS_CA: &str = "tls_ca";
+const CONFIG_NATS_TLS_CA_FILE: &str = "tls_ca_file";
+const CONFIG_NATS_MAX_WRITE_WAIT: &str = "max_write_wait";
+const CONFIG_NATS_STORAGE_MAX_AGE: &str = "max_age";
+const CONFIG_NATS_STORAGE_TYPE: &str = "storage_type";
+const CONFIG_NATS_STORAGE_NUM_REPLICAS: &str = "num_replicas";
+const CONFIG_NATS_STORAGE_COMPRESSION: &str = "compression";
+
+/// Configuration for connecting a NATS client.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct NatsConnectionConfig {
+    /// Cluster(s) to connect to
+    #[serde(default)]
+    pub cluster_uri: Option<String>,
+
+    /// JetStream Domain to connect to
+    #[serde(default)]
+    pub js_domain: Option<String>,
+
+    /// Auth JWT to use (if necessary)
+    #[serde(default)]
+    pub auth_jwt: Option<String>,
+
+    /// Auth seed to use (if necessary)
+    #[serde(default)]
+    pub auth_seed: Option<String>,
+
+    /// TLS Certificate Authority, encoded as a string
+    #[serde(default)]
+    pub tls_ca: Option<String>,
+
+    /// TLS Certificate Authority, as a path on disk
+    #[serde(default)]
+    pub tls_ca_file: Option<String>,
+
+    /// Backend storage configuration (opaque service provider specific configuration)
+    #[serde(default)]
+    pub storage_config: Option<StorageConfig>,
+
+    /// Write operation timeout configuration
+    pub max_write_wait: Option<u64>,
+}
+
+/// NATS Object-store is backed by its Key-Value store; so an object-store
+/// is identified by its backing bucket name.
+///
+/// Note that when storage config is provided via link configuration
+/// the following keys are expected:
+/// - `max_age` (optional): the maximum age of any object in the container, expressed in seconds; defaults to 10 years
+/// - `storage_type` (optional): the type of storage backend, File (default) and Memory
+/// - `num_replicas` (optional): how many replicas to keep for each object in a cluster, maximum 5; defaults to 1
+/// - `compression` (optional): whether the underlying stream should be compressed; defaults to false
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct StorageConfig {
+    /// Maximum age of any object in the container, expressed in nanoseconds
+    pub max_age: Option<core::time::Duration>,
+    /// Maximum size of the object store container, expressed in bytes
+    pub max_bytes: Option<i64>,
+    /// The type of storage backend, File (default) and Memory
+    pub storage_type: Option<StorageType>,
+    /// How many replicas to keep for each object in a cluster, maximum 5
+    pub num_replicas: Option<usize>,
+    /// Whether the underlying stream should be compressed
+    pub compression: Option<bool>,
```

Comment:
Same comment re: `#[serde(default)]`

Reply:
Added to make sure consistency is maintained.